components/server/src/auth/rate-limiter.ts

```diff
@@ -0,0 +1,212 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { GitpodServer } from "@gitpod/gitpod-protocol";
+import { log } from "@gitpod/gitpod-protocol/lib/util/logging";
+import { RateLimiterMemory, RateLimiterRes } from "rate-limiter-flexible";
+
+type GitpodServerMethodsType = keyof Omit<GitpodServer, "dispose" | "setClient">;
+type RateLimiterConfigType = { [K in GitpodServerMethodsType]: { type: MethodType, blocking: boolean } }
```

Comment:
Does it allow to configure only by type? I thought it would provide more fine grain control that we can fine tune during the incident.